src/src/facilitator/facilitator.service.ts

```diff
@@ -0,0 +1,255 @@
+          }
+        }
+      }`,
+      variables: variables
+    };
+
+    const response = await lastValueFrom(
+      this.httpService
+        .post(this.url, data, {
+          headers: {
+            'x-hasura-admin-secret': process.env.HASURA_ADMIN_SECRET,
+            'Content-Type': 'application/json',
+          },
+        })
+        .pipe(map((res) => res.data)),
+    );
+
+    let mappedResponse = response?.data?.core_faciltators;
+    
+    if (
+      mappedResponse
+      && body.hasOwnProperty('work_experience')
+      && this.isValidString(body.work_experience)
+    ) {
+      const isValidNumberFilter = !isNaN(parseInt(body.work_experience)) || body.work_experience === '5+';
+      if (isValidNumberFilter) {
+        mappedResponse = mappedResponse.filter(facilitator => {
+          if (facilitator.user?.experience && facilitator.user?.experience.length) {
+            const sum = facilitator.user?.experience.reduce((acc, curr) => {
+              if (curr.type === 'experience') {
+                acc += parseInt(curr.experience_in_years);
+              }
+              return acc;
+            }, 0);
+            if (body.work_experience === '5+' && sum > 5) {
+              return true;
+            } else if (parseInt(body.work_experience) === sum) {
+              return true;
+            } else {
+              return false;
+            }
+          } else {
+            return false;
+          }
+        });
+      }
+    }
+
+    if (
+      mappedResponse
+      && body.hasOwnProperty('vo_experience')
+      && this.isValidString(body.vo_experience)
+    ) {
+      const isValidNumberFilter = !isNaN(parseInt(body.vo_experience)) || body.vo_experience === '5+';
+      if (isValidNumberFilter) {
+        mappedResponse = mappedResponse.filter(facilitator => {
+          if (facilitator.user?.experience && facilitator.user?.experience.length) {
+            const sum = facilitator.user?.experience.reduce((acc, curr) => {
+              if (curr.type === 'vo_experience') {
+                acc += parseInt(curr.experience_in_years);
+              }
+              return acc;
+            }, 0);
+            if (body.vo_experience === '5+' && sum > 5) {
+              return true;
+            } else if (parseInt(body.vo_experience) === sum) {
+              return true;
+            } else {
+              return false;
+            }
+          } else {
+            return false;
+          }
+        });
+      }
+    }
+
+    let responseWithPagination = mappedResponse.slice(skip, skip + limit);
+
+    responseWithPagination = responseWithPagination.map(obj => {
+        const user_id = obj.user.id;
+        delete obj.user.id;
+        const res = {
+            ...obj,
+            'user_id': user_id,
+            ...obj.user     
+        };
+        delete res.user;
+        return res;
+    });
+
+    const count = mappedResponse.length;
+    const totalPages = Math.ceil(count / limit);
+
+    return {
```

Comment:
write response in single format & maintain that: 
instead of this: return {
      statusCode: 200,
      message: 'Ok.',
      totalCount: count,
      data: responseWithPagination,
      limit,
      currentPage: page,
      totalPages: `${totalPages}`,
    };
return this:
return response.status(200).json({
      message: 'wite proper message here',
      data:{totalCount: count,
      data: responseWithPagination,
      limit,
      currentPage: page,
      totalPages: `${totalPages}`,}
    })

Reply:
Done